src/main/java/usecases/disconnecting/DcInteractor.java

```diff
@@ -0,0 +1,78 @@
+package usecases.disconnecting;
+
+import entities.LobbyManager;
+import entities.Player;
+import exceptions.PlayerNotFoundException;
+import usecases.Response;
+import usecases.disconnecting.boundaries.DcInputBoundary;
+import usecases.disconnecting.boundaries.DcOutputBoundary;
+import usecases.disconnecting.data.DcInputData;
+import usecases.disconnecting.data.DcOutputData;
+
+/**
+ * Interactor for Disconnecting Use Case
+ */
+public class DcInteractor implements DcInputBoundary {
+    private final LobbyManager lm;
+    DcOutputBoundary dcOutputBoundary;
+
+    /**
+     * Constructor for DcInteractor
+     * @param lm Lobby Manager
+     * @param dcOutputBoundary DcOutputBoundary
+     */
+    public DcInteractor(LobbyManager lm, DcOutputBoundary dcOutputBoundary) {
+        this.lm = lm;
+        this.dcOutputBoundary = dcOutputBoundary;
+    }
+
+    /**
+     * Disconnects the user
+     * @param data input data which contains playerId
+     */
+    @Override
+    public void disconnect(DcInputData data) {new Thread(new DcThread(data.playerId)).start();}
+
+    /**
+     * Thread for disconnecting the player
+     */
+    public class DcThread implements Runnable {
+        private final String playerId;
+        private Player playerToDisconnect;
+
+        /**
+         * Constructor for Disconnecting Thread
+         * @param playerId ID of the player we need to disconnect
+         */
+        public DcThread(String playerId) {this.playerId = playerId;}
+
+        @Override
+        public void run() {
+            Response response = new Response(Response.ResCode.SUCCESS, "Disconnecting was successful.");
+            try {
+                if(playerIsInThePool(playerId))
+                    lm.removeFromPoolCancel(playerToDisconnect);
+            } catch (PlayerNotFoundException e) {
+                response = new Response(Response.ResCode.PLAYER_NOT_FOUND, "Disconnecting was not successful. User was not found.");
+                e.printStackTrace();
+            }
+            DcOutputData outputData = new DcOutputData(response);
+            dcOutputBoundary.hasDisconnected(outputData);
```

Comment:
Missing second branch of logic. This use case should check if the player is in the pool and then remove them from the pool if they are. If the player was not in the pool, then it should check if the player is in the game and remove them from the game if they are. Finally, if player was not found in either, then return PlayerNotFound. If player was found and removed, return success